# Source: https://gitlab.com/gitlab-org/gitlab-foss/-/blob/v18.11.11/Gemfile
# CVE-2025-67202 requires sidekiq-cron 2.4.0: https://nvd.nist.gov/vuln/detail/CVE-2025-67202
# GHSA-3h96-34p3-xm76 requires graphql 2.5.26: https://github.com/advisories/GHSA-3h96-34p3-xm76
# sys-filesystem ~> 1.5: 1.4.3 FFI-attaches statvfs64, which musl does not export
# CVE-2026-63435 requires mail 2.9.1: https://nvd.nist.gov/vuln/detail/CVE-2026-63435
# sidekiq-cron 2.4 compatibility: https://github.com/sidekiq-cron/sidekiq-cron/compare/v2.3.1...v2.4.0
diff --git a/Gemfile b/Gemfile
--- a/Gemfile
+++ b/Gemfile
@@ -166,7 +166,7 @@
 gem 'rack-cors', '~> 2.0.1', require: 'rack/cors', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
 
 # GraphQL API
-gem 'graphql', '2.5.11', feature_category: :api
+gem 'graphql', '~> 2.5.26', feature_category: :api
 gem 'graphql-docs', '~> 5.2.0', group: [:development, :test], feature_category: :api
 gem 'apollo_upload_server', '~> 2.1.6', feature_category: :api
 
@@ -297,7 +297,7 @@
 
 # Background jobs
 gem 'sidekiq', path: 'vendor/gems/sidekiq', require: 'sidekiq', feature_category: :scalability
-gem 'sidekiq-cron', '~> 2.3.0', feature_category: :scalability
+gem 'sidekiq-cron', '~> 2.4.0', feature_category: :scalability
 gem 'gitlab-sidekiq-fetcher',
   path: 'vendor/gems/sidekiq-reliable-fetch',
   require: 'sidekiq-reliable-fetch',
@@ -632,7 +632,7 @@
 
 # System information
 gem 'vmstat', '~> 2.3.0', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
-gem 'sys-filesystem', '~> 1.4.3', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
+gem 'sys-filesystem', '~> 1.5', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
 
 # NTP client
 gem 'net-ntp', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
@@ -690 +690 @@
-gem 'mail', '= 2.9.0', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
+gem 'mail', '= 2.9.1', feature_category: :shared # rubocop:todo Gemfile/MissingFeatureCategory -- https://gitlab.com/gitlab-org/gitlab/-/issues/581839
diff --git a/lib/gitlab/patch/sidekiq_cron_poller.rb b/lib/gitlab/patch/sidekiq_cron_poller.rb
--- a/lib/gitlab/patch/sidekiq_cron_poller.rb
+++ b/lib/gitlab/patch/sidekiq_cron_poller.rb
@@ -13 +13 @@
-if Gem::Version.new(Sidekiq::Cron::VERSION) != Gem::Version.new('2.3.1')
+if Gem::Version.new(Sidekiq::Cron::VERSION) != Gem::Version.new('2.4.0')
